src/simplify_reshapes.cpp

```diff
@@ -176,6 +176,23 @@ struct find_nested_convert
     }
 };
 
+struct find_useless_convert
+{
+    auto matcher() const { return match::name("convert"); }
+
+    void apply(module& m, const match::matcher_result& mr) const
+    {
+        auto ins     = mr.result;
+        auto x       = ins->inputs().front();
+        auto outputs = x->outputs();
+        if(x->get_shape() == ins->get_shape())
+        {
+            m.replace_instruction(ins, x);
+            m.move_instruction(ins, m.end());
+        }
+    }
+};
```

Comment:
The `find_nop_reshapes` should already handle this case.

Reply:
removed